### ISSUE TITLE
Solved quests get attributed to the wrong country

### ISSUE THREAD

**Issue by rhhsm** [streetcomplete/StreetComplete#2987]
Last week, I travelled from Bulgaria to the Netherlands for the first time since corona and since I started using the app. I solved some quest around Sofia and Eindhoven airports, and then uploaded the solutions through the airport bus wifi. To my surprise, I found that I now had 20 stars for solving quests in Romania! Thinking that a batch of quest solutions may be attributed to the country in which the geographic average of their location is located, I answered some further quests in the Netherlands and a few in Bulgaria that I happened to know the answer to, hoping to get some credits for Germany or Austria. I was even more surprised to get an additional 16 stars for Romania as well as 30 for Ukraine... Not a high priority bug, but intriguing.

**How to Reproduce**
With automatic upload of answers off, answer a few quests in widely spaced countries, and then upload them manually as a batch.

**Versions affected**
Android 10, SC 32.1

**Comment by westnordost** [comment]
Yes, that's the current behavior and not a bug.

On June 21, 2021 9:41:54 AM GMT+02:00, Richard ***@***.***> wrote:
>Last week, I travelled from Bulgaria to the Netherlands for the first
>time since corona and since I started using the app. I solved some
>quest around Sofia and Eindhoven airports, and then uploaded the
>solutions through the airport bus wifi. To my surprise, I found that I
>now had 20 stars for solving quests in Romania! Thinking that a batch
>of quest solutions may be attributed to the country in which the
>geographic average of their location is located, I answered some
>further quests in the Netherlands and a few in Bulgaria that I happened
>to know the answer to, hoping to get some credits for Germany or
>Austria. I was even more surprised to get an additional 16 stars for
>Romania as well as 30 for Ukraine... Not a high priority bug, but
>intriguing.
>
>**How to Reproduce**
>With automatic upload of answers off, answer a few quests in widely
>spaced countries, and then upload them manually as a batch.
>
>**Versions affected**
>Android 10, SC 32.1
>
>
>
>-- 
>You are receiving this because you are subscribed to this thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/streetcomplete/StreetComplete/issues/2987

-- 
Diese Nachricht wurde von meinem Android-Gerät mit K-9 Mail gesendet.

**Comment by matkoniecz** [comment]
note that for example HDYC ( https://hdyc.neis-one.org/ ) is working in the same way - calculating center of bbox, getting number of edits is very easy.

Correctly attributing elements in large bbox requires much, much, much more intensive, complicated and trickier processing.

It can be avoided by pushing changes before moving to a far away location.

Feel free to enjoy extra flags :)

**Comment by rhhsm** [comment]
> 
> Yes, that's the current behavior and not a bug.

Well, it's not how it should behave, isn't it, so it is a bug. But I can imagine no one would want to spend time on it. It creates new ways of becoming no. 1 in North Korea!

**Comment by matkoniecz** [comment]
It is happening this way by design, as alternative is N times more complex, complicated and expensive to host.

When "works as expected, but mismatches expectations" is a bug and when is not is kind of definitional issue.

**Comment by legofahrrad** [comment]
I had to noticed this behaviour too, when I was in Rome and Vatican city this week. I edited in both countries and had 24 edits in Vatican city in between. After solving some other quests in Rome, 20 quests were missing in Vatican city and counted to Italy then. So I would be happy if the country detection would work better than just looking at the bbox's center. This is especially for these "small countries" helpful, such as Vatican city or Monaco but of course also for border regions.
